clockservice.js

```diff
@@ -1,11 +1,18 @@
 'use strict';
 
 const
-  dateformat = require('dateformat'),
-  vscode = require('vscode');
+    dateformat = require('dateformat'),
+    vscode = require('vscode');
 
 const DEFAULT_DATE_FORMAT = 'hh:MM TT';
+const DEFAULT_INSERT_FORMAT = 'dd/mm/yyyy hh:MM TT';
 
-module.exports = function () {
-  return dateformat(Date.now(), vscode.workspace.getConfiguration('clock').dateFormat || DEFAULT_DATE_FORMAT);
-};
+module.exports = {
+    insert: function() {
+        return dateformat(Date.now(), vscode.workspace.getConfiguration('clock').insertFormat || DEFAULT_INSERT_FORMAT);
```

Comment:
If you fall back to `dateFormat` when `insertFormat` is not present, then you get the same behaviour as the current version when there is only 1 format set.
```suggestion
        return dateformat(Date.now(), vscode.workspace.getConfiguration('clock').insertFormat || vscode.workspace.getConfiguration('clock').dateFormat || DEFAULT_INSERT_FORMAT);
```